libs/workspace-manager/src/lib/features/connect-workspace/connect-workspace.component.ts

```diff
@@ -1,36 +1,109 @@
+import {
+  CURRENT_WORKSPACE_PATH,
+  Directory,
+} from '@angular-cli-gui/shared/data';
+import {
+  BehaviorSubject,
+  combineLatest,
+  of,
+  Subject,
+  switchMap,
+  takeUntil,
+} from 'rxjs';
 
 import { ConnectWorkspaceService } from '../../data-access/connect-workspace.service';
+import { WorkspaceManagerApiService } from '../../data-access/workspace-manager-api.service';
+import { FilesystemNavigatorComponent } from '../../ui';
 
 @Component({
   selector: 'cli-connect-workspace',
   standalone: true,
-  imports: [CommonModule, MatInputModule, MatButtonModule, ReactiveFormsModule],
+  imports: [CommonModule, MatButtonModule, FilesystemNavigatorComponent],
   templateUrl: './connect-workspace.component.html',
   styleUrls: ['./connect-workspace.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConnectWorkspaceComponent {
+export class ConnectWorkspaceComponent implements OnInit, OnDestroy {
   connectService = inject(ConnectWorkspaceService);
-  form = new FormGroup(
-    {
-      path: new FormControl('', [Validators.required]),
-    },
-    { updateOn: 'submit' }
-  );
+  workspaceManagerApiService = inject(WorkspaceManagerApiService);
+
+  directories$ = new BehaviorSubject<Directory[]>([]);
+  path$ = new BehaviorSubject<string | null>(null);
+  pathSeparator$ = new BehaviorSubject<string | null>(null);
+  isAngularWorkspace$ = new BehaviorSubject<boolean>(false);
+  private destroyed$ = new Subject<boolean>();
+
+  ngOnInit(): void {
+    this.workspaceManagerApiService
+      .getPathSeparator()
+      .subscribe((separator) => {
+        this.pathSeparator$.next(separator);
+        this.initWorkspacePath();
+      });
+
+    combineLatest([this.path$, this.pathSeparator$])
+      .pipe(
+        switchMap(([path, separator]) => {
+          path && separator && this.setIsAngularWorkspace(path, separator);
+          return path
+            ? this.workspaceManagerApiService.getDirectoriesInPath(path)
+            : of([]);
+        }),
+        takeUntil(this.destroyed$)
+      )
+      .subscribe((directories) => {
+        this.directories$.next(directories);
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.destroyed$.next(true);
+  }
 
   connectWorkspace(): void {
-    if (!this.form.valid) return;
-    const path = this.form.controls.path.value ?? '';
-    this.connectService.connectWorkspace(path).subscribe();
+    const path = this.path$.getValue();
+    if (path) {
+      this.connectService.connectWorkspace(path).subscribe();
+    }
+  }
+
+  onPathChanged(path: string): void {
+    if (path !== this.path$.getValue()) {
+      this.path$.next(path);
+    }
+  }
+
+  private initWorkspacePath(): void {
+    const currentWorkspacePath = sessionStorage.getItem(CURRENT_WORKSPACE_PATH);
+    const path$ = currentWorkspacePath
+      ? of(currentWorkspacePath)
+      : this.workspaceManagerApiService.getHomeDir();
+    path$.subscribe((path: string) => this.path$.next(path));
```

Comment:
Don't you need to handle unsubscribing?

Reply:
should you unsubscribe an `of` observable ?
the second one is an http observable